### PR TITLE
Creates the InputNumber component

### PR DESCRIPTION
**Pull request by clayton-duarte**
- Creates the InputNumber component;
- Simplifies the Input component;

### CODE REVIEW

File: src/components/Input/index.tsx [hunk 1]
```diff
@@ -36,33 +35,19 @@ const StyledInput = styled.input<{ hasError?: boolean }>`
...
-  onBlur,
-  hasError,
-  onFocus,
-  name,
-  ...inputProps
+  InputHTMLAttributes<HTMLInputElement>> = ({
+  name = 'text-input',
+  hasError = false,
+  type = 'text',
+  ...inputProps // InputHTMLAttributes<HTMLInputElement>
```
Comment (l0rdcafe) [inline src/components/Input/index.tsx:42]: question: is this where onBlur etc... hooks were moved?

File: src/components/InputNumber/index.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import ThemeProvider from '../..'
+import InputNumber from '.'
+
+describe('InputNumber', () => {
+  it('should render properly', () => {
+    const { container } = render(
+      <ThemeProvider>
+        <InputNumber name="InputNumber" />
+      </ThemeProvider>
+    )
```
Comment (l0rdcafe) [inline src/components/InputNumber/index.spec.tsx:13]: there's a lot of logic in this component, can we add some tests?

File: src/components/Input/index.tsx (hunk 1 above)
Comment (clayton-duarte) [reply src/components/Input/index.tsx:42]: hehe, yup. I removed all the redundant props declaration

File: src/components/InputNumber/index.mdx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+import InputNumber from '.'
+
+# InputNumber
+
+### Use Cases
+
+##### Default
+
+<InputNumber name="my-InputNumber" />
```
Comment (l0rdcafe) [inline src/components/InputNumber/index.mdx:16]: i'm wondering if down the line we could have 1 outward facing Input component that takes a variant whether it's `number` `text` `email` `password` `sensitive` etc...?

File: src/components/InputNumber/index.tsx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+  FunctionComponent,
+  ChangeEvent,
+  useEffect,
+  useState,
+  useMemo,
+} from 'react'
+
+import Input, { InputProps } from '../Input'
+
+export interface ReturnValue {
```
Comment (l0rdcafe) [inline src/components/InputNumber/index.tsx:12]: trying to think of a better name `Output` but unsure tbh haha

**Review by l0rdcafe — APPROVED** [review]

File: src/components/InputNumber/index.mdx (hunk 3 above)
Comment (clayton-duarte) [reply src/components/InputNumber/index.mdx:16]: thats a good suggestion, like, we can have a BaseInput, Some variations extending it, and a root Input component that receives a variant prop to select them

**Pull request merged**